mysqld_exporter.go

```diff
@@ -1643,6 +1666,33 @@ func scrapeTableSchema(db *sql.DB, ch chan<- prometheus.Metric) error {
 	}
 
 	for _, database := range dbList {
+		if _, ok := tableSchemaLastRun[database]; !ok {
+			tableSchemaLastRun[database] = time.Time{}
+		}
+		random := rand.Float64() * 0.25
+		nextRun := *tableSchemaAutoAnalyzeDuration + time.Duration(float64(*tableSchemaAutoAnalyzeDuration)*random)
+		if *tableSchemaAutoAnalyze && time.Since(tableSchemaLastRun[database]) > nextRun {
```

Comment:
What @fabxc meant above (in case that isn't clear anyway): This line will work just fine because for unknown databases, the zero value of time.Time is returned (which is very far in the past), so the analyze will be run on new databases (which, actually, with what you are doing above, it will only run 5min later...)